### PR TITLE
Feature/pp9 8341 align sdk contract metadata type

### PR DESCRIPTION
**Pull request by phoeniks-sk**

### CODE REVIEW

File: src/Picturepark.SDK.V1.Contract/Contract/ContentItem.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ public class ContentItem<T>
         public System.Collections.Generic.ICollection<string> LayerSchemaIds { get; set; } = new System.Collections.Generic.List<string>();
 
         /// <summary>The metadata dictionary</summary>
-        public DataDictionary Metadata { get; set; }
+        public object Metadata { get; set; }
```
Comment (phoeniks-sk) [inline src/Picturepark.SDK.V1.Contract/Contract/ContentItem.cs]: Change to IDictionary<string, object>

File: src/Picturepark.SDK.V1.Contract/Contract/Metadata.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace Picturepark.SDK.V1.Contract
+{
+    public static class Metadata
+    {
+        public static IDictionary<string, object> From(params object[] layers)
```
Comment (phoeniks-sk) [inline src/Picturepark.SDK.V1.Contract/Contract/Metadata.cs]: Add comments

File: src/Picturepark.SDK.V1.Contract/Contract/Metadata.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        {
+            return layers.ToDictionary(l => l.GetType().Name.ToLowerCamelCase());
```
Comment (phoeniks-sk) [inline src/Picturepark.SDK.V1.Contract/Contract/Metadata.cs]: Use the Schema id property on the attribute if provided

File: src/Picturepark.SDK.V1.Contract/Partials/ContentDetail.cs [hunk 4]
```diff
@@ -33,8 +38,32 @@ public ContentItem<T> AsContentItem<T>()
...
-                Content = item
+                Content = ContentAs<T>()
             };
         }
+
+        public T ContentAs<T>() => Content is T content ? content : ((JObject)Content).ToObject<T>();
+
+        public T Layer<T>(string name = null)
+        {
+            var layer = Layer(name ?? typeof(T).Name);
```
Comment (phoeniks-sk) [inline src/Picturepark.SDK.V1.Contract/Partials/ContentDetail.cs]: Use the schema id on the attribute if provided

File: src/Picturepark.SDK.V1.Contract/Extensions/MetadataExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+﻿using System.Collections.Generic;
+
+namespace Picturepark.SDK.V1.Contract.Extensions
+{
+    public static class MetadataExtensions
+    {
+        public static IDictionary<string, object> AsMetadata(this object obj) => Metadata.From(obj);
```
Comment (phoeniks-sk) [inline src/Picturepark.SDK.V1.Contract/Extensions/MetadataExtensions.cs]: Add optional parameter to change name

File: src/Picturepark.SDK.V1.Contract/Partials/ContentDetail.cs [hunk 6]
```diff
@@ -33,8 +38,32 @@ public ContentItem<T> AsContentItem<T>()
...
+
+        public bool HasLayer(string name) => JMetadata.ContainsKey(name.ToLowerCamelCase());
+
+        public DisplayValueDictionary LayerDisplayValues<T>()
+            => LayerDisplayValues(typeof(T).Name);
+
+        public DisplayValueDictionary LayerDisplayValues(string name)
+        {
+            var layer = Layer(name);
+            return layer.GetValue("_displayValues").ToObject<DisplayValueDictionary>();
```
Comment (phoeniks-sk) [inline src/Picturepark.SDK.V1.Contract/Partials/ContentDetail.cs]: Check for existence of _displayValues

File: src/Picturepark.SDK.V1.Contract/Partials/ContentMetadataUpdateRequest.cs [hunk 7]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Linq.Expressions;
+
+namespace Picturepark.SDK.V1.Contract
+{
+    public partial class ContentMetadataUpdateRequest
+    {
+        public static ContentMetadataUpdateRequest LayerMergeUpdate<T, TProperty>(Expression<Func<T, TProperty>> property, TProperty newValue)
```
Comment (phoeniks-sk) [inline src/Picturepark.SDK.V1.Contract/Partials/ContentMetadataUpdateRequest.cs]: Add comment

File: src/Picturepark.SDK.V1.Contract/Partials/DisplayValueDictionary.cs [hunk 8]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Picturepark.SDK.V1.Contract
+{
+    public partial class DisplayValueDictionary
+    {
+        public string Name => this["name"];
```
Comment (phoeniks-sk) [inline src/Picturepark.SDK.V1.Contract/Partials/DisplayValueDictionary.cs]: Add others

File: src/Picturepark.SDK.V1.Tests/Contracts/AllDataTypesContract.cs [hunk 9]
```diff
@@ -42,7 +42,7 @@ public class AllDataTypesContract
 
         public GeoPoint GeoPointField { get; set; }
 
-        public int IntegerField { get; set; }
+        public int? IntegerField { get; set; }
```
Comment (phoeniks-sk) [inline src/Picturepark.SDK.V1.Tests/Contracts/AllDataTypesContract.cs]: Change also other value types to Nullable<>

File: src/Picturepark.SDK.V1/Conversion/ClassToSchemaConverter.cs [hunk 10]
```diff
@@ -32,7 +32,7 @@ public ClassToSchemaConverter(IContractResolver contractResolver)
 
         public static string ResolveSchemaName(Type contract)
```
Comment (phoeniks-sk) [inline src/Picturepark.SDK.V1/Conversion/ClassToSchemaConverter.cs]: ResolveSchemaId

File: src/Picturepark.SDK.V1.Contract/Partials/ContentDetail.cs [hunk 11]
```diff
@@ -33,8 +32,32 @@ public ContentItem<T> AsContentItem<T>()
...
+            => Metadata.TryGetValue(name.ToLowerCamelCase(), out var layer) ? (JObject)layer : null;
+
+        public bool HasLayer<T>() => HasLayer(Contract.Metadata.ResolveLayerKey(typeof(T)));
+
+        public bool HasLayer(string name) => Metadata.ContainsKey(name.ToLowerCamelCase());
+
+        public DisplayValueDictionary LayerDisplayValues<T>()
+            => LayerDisplayValues(Contract.Metadata.ResolveLayerKey(typeof(T)));
+
+        public DisplayValueDictionary LayerDisplayValues(string name)
```
Comment (simobeloz) [inline src/Picturepark.SDK.V1.Contract/Partials/ContentDetail.cs]: Add comments to methods

File: src/Picturepark.SDK.V1.Contract/Partials/DisplayValueDictionary.cs [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+{
+    public partial class DisplayValueDictionary
+    {
+        public string Name => this[nameof(DisplayPatternType.Name).ToLowerCamelCase()];
+
+        public string Detail => this[nameof(DisplayPatternType.Detail).ToLowerCamelCase()];
+
+        public string List => this[nameof(DisplayPatternType.List).ToLowerCamelCase()];
+
+        public string Thumbnail => this[nameof(DisplayPatternType.Thumbnail).ToLowerCamelCase()];
```
Comment (simobeloz) [inline src/Picturepark.SDK.V1.Contract/Partials/DisplayValueDictionary.cs]: Add comments

**Pull request merged**